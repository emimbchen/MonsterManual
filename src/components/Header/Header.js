import { NavLink } from "react-router-dom";
import { ReactComponent as CardIcon } from "../../assets/cards.svg";

function Header({ ...props }) {
  return (
    <header role="banner" className="shadow" {...props}>
      <div className="container py-4 flex items-center justify-center">
        <NavLink to={`/`} end>
          <CardIcon className="h-12" alt="Spell Card Logo" />
        </NavLink>
      </div>
    </header>
  );
}

export default Header;
