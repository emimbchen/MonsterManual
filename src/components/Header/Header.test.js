import axe from "axe-core";
import Header from "./Header";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";

const mockHeader = (
  <MemoryRouter>
    <Header />
  </MemoryRouter>
);

describe("Header Component", () => {
  test("passes accessibility check", async () => {
    const { container } = render(mockHeader);
    const results = await axe.run(container);
    expect(results.violations.length).toBe(0);
  });

  it("renders correctly", () => {
    const { asFragment } = render(mockHeader);
    expect(asFragment()).toMatchSnapshot();
  });
});
