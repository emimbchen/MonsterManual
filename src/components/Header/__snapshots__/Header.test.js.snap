// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component renders correctly 1`] = `
<DocumentFragment>
  <header
    class="shadow"
    role="banner"
  >
    <div
      class="container py-4 flex items-center justify-center"
    >
      <a
        aria-current="page"
        aria-label="Navigate home"
        class="active"
        href="/"
      >
        <svg
          alt="Spell Card Logo"
          class="h-12"
        >
          cards.svg
        </svg>
      </a>
    </div>
  </header>
</DocumentFragment>
`;
